server/manager/src/main/java/org/apache/accumulo/manager/tableOps/compact/CompactionDriver.java

```diff
@@ -241,17 +242,27 @@ public int updateAndCheckTablets(Manager manager, FateId fateId)
             noneSelected++;
           } else {
             var mutator = tabletsMutator.mutateTablet(tablet.getExtent()).requireAbsentOperation()
-                .requireSame(tablet, FILES, SELECTED, ECOMP, COMPACTED);
-            var selectedFiles =
-                new SelectedFiles(filesToCompact, tablet.getFiles().equals(filesToCompact), fateId);
+                .requireSame(tablet, FILES, SELECTED, ECOMP, COMPACTED, USER_COMPACTION_REQUESTED);
+            var selectedFiles = new SelectedFiles(filesToCompact,
+                tablet.getFiles().equals(filesToCompact), fateId, time);
 
             mutator.putSelectedFiles(selectedFiles);
 
+            // We no longer need to include this marker if files are selected
+            if (tablet.getUserCompactionsRequested().contains(fateId)) {
+              mutator.deleteUserCompactionRequested(fateId);
+            }
+
             selectionsSubmitted.put(tablet.getExtent(), filesToCompact);
 
+            // TODO: Do we need to handle a race condition for the rejection handler check
```

Comment:
Re this comment.  Could relax the check in the rejection handler to `tabletMetadata.getSelectedFiles().getFateId().equals(fateId) || tabletMetadata.getCompacted().contains(fateId)`  if either of those are true then write when through, but could have changed since to compact some or even completed.

Reply:
Alright so the full rejection handler should just look like the following? Just want to make sure i have it right.

```java
mutator.submit(tabletMetadata -> tabletMetadata.getSelectedFiles() != null
  && tabletMetadata.getSelectedFiles().getFateId().equals(fateId) ||
  tabletMetadata.getCompacted().contains(fateId));
```